src/utils/SiriusAdapter.cpp

```diff
@@ -386,16 +405,42 @@ class TOPPSiriusAdapter :
     siriusfile.store(out_sirius, sirius_result);
 
     // convert sirius_output to mztab and store file
-    if (out_csifingerid.empty() == false)
+    if (!out_csifingerid.empty())
     {
       MzTab csi_result;
       MzTabFile csifile;
       CsiFingerIdMzTabWriter::read(subdirs, in, top_n_hits, csi_result);
       csifile.store(out_csifingerid, csi_result);
     }
 
+    // should the sirius workspace be retained
+    if(!sirius_workspace_directory.empty())
+    {
+      // convert path to absolute path
+      QDir sw_dir(sirius_workspace_directory.toQString());
+      sirius_workspace_directory = String(sw_dir.absolutePath());
+      
+      // try to create directory if not present
+      if (!sw_dir.exists())
+      {
+        sw_dir.mkpath(sirius_workspace_directory.toQString());
+      }
+      
+      // move tmp folder to new location
+      std::rename(tmp_dir.toStdString().c_str(), sirius_workspace_directory.c_str());
+      LOG_WARN << "Sirius Workspace was moved to " << sirius_workspace_directory << std::endl;
+    }
+   
+    // should the ms file be retained (non-converter mode)
+    if(!out_ms.empty())
+    {  
+      QFile::copy(tmp_ms_file.toQString(), out_ms.toQString());
+      LOG_WARN << "Preprocessed .ms files was moved to " << out_ms << std::endl; 
+    }
+
+
     // clean tmp directory if debug level < 2
-    if (debug_level_ >= 2)
+    if (debug_level_ >= 2 && out_ms.empty() && sirius_workspace_directory.empty())
```

Comment:
why will it only be kept if out_ms == "" and sirius_workspace_directory == ""?
because it was already moved? please document this